### PR TITLE
[SPARK-26921[R][DOCS] Document Arrow optimization and vectorized R APIs

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR adds SparkR with Arrow optimization documentation.

Note that looks CRAN issue in Arrow side won't look likely fixed soon, IMHO, even after Spark 3.0.
If it happen to be fixed, I will fix this doc too later.

Another note is that Arrow R package itself requires R 3.5+. So, I intentionally didn't note this.

## How was this patch tested?

Manually built and checked.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @felixcheung, @BryanCutler, @viirya

**Review by srowen — APPROVED** [review]

File: docs/sparkr.md [hunk 1]
```diff
@@ -657,6 +657,65 @@ The following example shows how to save/load a MLlib model by SparkR.
...
+
+## Ensure Arrow Installed
+
+Currently, Arrow R library is not on CRAN yet [ARROW-3204](https://issues.apache.org/jira/browse/ARROW-3204). Therefore, it should be installed directly from Github. You can use `remotes::install_github` as below.
+
+```bash
+Rscript -e 'remotes::install_github("apache/arrow@TAG", subdir = "r")'
+```
+
+`TAG` is a version tag that can be checked in [Arrow at Github](https://github.com/apache/arrow/releases). You must ensure that Arrow R packge is installed and available on all cluster nodes. The current supported version is 0.12.0.
```
Comment (BryanCutler) [inline docs/sparkr.md]: should it be version 0.12.1 to match Python?

Comment (HyukjinKwon) [reply docs/sparkr.md]: Yup.

**Comment by HyukjinKwon** [comment]
Merged to master.

It looked a bit odd because I squashed the last commit but the actual diff the current Jenkins is testing is https://github.com/apache/spark/pull/24506#discussion_r280243059 - we don't build the doc in PR builder.

File: docs/sparkr.md [hunk 2]
```diff
@@ -657,6 +657,65 @@ The following example shows how to save/load a MLlib model by SparkR.
...
+# Apache Arrow in SparkR
+
+Apache Arrow is an in-memory columnar data format that is used in Spark to efficiently transfer data between JVM and R processes. See also PySpark optimization done, [PySpark Usage Guide for Pandas with Apache Arrow](sql-pyspark-pandas-with-arrow.html). This guide targets to explain how to use Arrow optimization in SparkR with some key points.
+
+## Ensure Arrow Installed
+
+Currently, Arrow R library is not on CRAN yet [ARROW-3204](https://issues.apache.org/jira/browse/ARROW-3204). Therefore, it should be installed directly from Github. You can use `remotes::install_github` as below.
+
+```bash
+Rscript -e 'remotes::install_github("apache/arrow@TAG", subdir = "r")'
```
Comment (felixcheung) [inline docs/sparkr.md]: I think it will be better to give the full command - ie. TAG as 0.12.1

**Review by felixcheung — COMMENTED** [review]
this looks really good. thx

File: docs/sparkr.md (hunk 2 above)
Comment (HyukjinKwon) [reply docs/sparkr.md]: Hm, yea .. let me fix it later when I touch this before 3.0.

**Pull request closed without merging**